tests/saw/spec/handshake/s2n_handshake_io.cry

```diff
@@ -943,7 +977,11 @@ valid_handshake hs = handshakefn != zero /\
                      handshakefn = (handshakes_fn hs.handshake_type)
 
 valid_connection : connection -> Bit
-valid_connection conn = valid_handshake (conn.handshake)
+valid_connection conn = valid_handshake (conn.handshake) /\
```

Comment:
Is there any way to simplify this expression or otherwise make it easier to understand? Maybe it would be more clear if the spec was written in a more general way. This form of the spec seems to depend on the fact that 1.2 and 1.3 are the only TLS versions supported.

Reply:
The confusion here might be my fault. We only have two state machines, one to do TLS13 and one to do <=TLS12. But we usually refer to them as the TLS12 or TLS13 state machine, even though other versions can be negotiated in the "TLS12 state machine".

Reply:
Yan added some comments here that might help the confusion.